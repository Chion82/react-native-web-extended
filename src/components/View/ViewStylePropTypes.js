import BorderPropTypes from '../../propTypes/BorderPropTypes';
import ColorPropType from '../../propTypes/ColorPropType';
import LayoutPropTypes from '../../propTypes/LayoutPropTypes';
import { PropTypes } from 'react';
import ShadowPropTypes from '../../propTypes/ShadowPropTypes';
import TransformPropTypes from '../../propTypes/TransformPropTypes';

const { number, oneOf, string } = PropTypes;
const autoOrHiddenOrVisible = oneOf([ 'auto', 'hidden', 'visible' ]);
const hiddenOrVisible = oneOf([ 'hidden', 'visible' ]);

module.exports = process.env.NODE_ENV !== 'production' ? {
  ...BorderPropTypes,
  ...LayoutPropTypes,
  ...ShadowPropTypes,
  ...TransformPropTypes,
  backfaceVisibility: hiddenOrVisible,
  backgroundColor: ColorPropType,
  opacity: number,
  overflow: autoOrHiddenOrVisible,
  zIndex: number,
  /*
   * @platform web
   */
  backgroundAttachment: string,
  backgroundClip: string,
  backgroundImage: string,
  backgroundPosition: string,
  backgroundOrigin: oneOf([ 'border-box', 'content-box', 'padding-box' ]),
  backgroundRepeat: string,
  backgroundSize: string,
  boxShadow: string,
  cursor: string,
  outline: string,
  overflowX: autoOrHiddenOrVisible,
  overflowY: autoOrHiddenOrVisible,
  transitionDelay: string,
  transitionDuration: string,
  transitionProperty: string,
  transitionTimingFunction: string,
  userSelect: string,
  visibility: hiddenOrVisible,
  WebkitOverflowScrolling: oneOf([ 'auto', 'touch' ])
} : {};
