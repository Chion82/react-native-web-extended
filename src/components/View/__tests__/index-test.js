/* eslint-env jasmine, jest */

import React from 'react';
import renderer from 'react-test-renderer';
import View from '../';

jest.mock('react-dom');

describe('components/View', () => {
  describe('rendered element', () => {
    test('is a "div" by default', () => {
      const component = renderer.create(<View />);
      expect(component.toJSON()).toMatchSnapshot();
    });

    test('is a "span" when inside <View accessibilityRole="button" />', () => {
      const component = renderer.create(<View accessibilityRole='button'><View /></View>);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  test('prop "children"', () => {
    const children = <View testID='1' />;
    const component = renderer.create(<View>{children}</View>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('prop "pointerEvents"', () => {
    const component = renderer.create(<View pointerEvents='box-only' />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('prop "style"', () => {
    let component = renderer.create(<View />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<View style={{ flex: 1 }} />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<View style={{ flexShrink: 1 }} />);
    expect(component.toJSON()).toMatchSnapshot();

    component = renderer.create(<View style={{ flex: 1, flexShrink: 2 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
