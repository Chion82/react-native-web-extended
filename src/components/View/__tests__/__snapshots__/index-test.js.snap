exports[`components/View prop "children" 1`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  }>
  <div
    className=" __style_df"
    data-testid="1"
    style={
      Object {
        "MozBoxSizing": "border-box",
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "inherit",
        "display": null,
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "font": "inherit",
        "listStyle": "none",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "textAlign": "inherit",
        "textDecoration": "none",
      }
    } />
</div>
`;

exports[`components/View prop "pointerEvents" 1`] = `
<div
  className=" __style_df __style_pebo"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "pointerEvents": null,
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View prop "style" 1`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View prop "style" 2`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 1,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 1,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 1,
      "msFlexPositive": 1,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View prop "style" 3`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 1,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 1,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 1,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View prop "style" 4`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexGrow": 1,
      "WebkitFlexShrink": 2,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexGrow": 1,
      "flexShrink": 2,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 2,
      "msFlexPositive": 1,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View rendered element is a "div" by default 1`] = `
<div
  className=" __style_df"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  } />
`;

exports[`components/View rendered element is a "span" when inside <View accessibilityRole="button" /> 1`] = `
<button
  className=" __style_df"
  role="button"
  style={
    Object {
      "MozBoxSizing": "border-box",
      "WebkitAlignItems": "stretch",
      "WebkitBoxAlign": "stretch",
      "WebkitBoxDirection": "normal",
      "WebkitBoxOrient": "vertical",
      "WebkitFlexBasis": "auto",
      "WebkitFlexDirection": "column",
      "WebkitFlexShrink": 0,
      "alignItems": "stretch",
      "backgroundColor": "transparent",
      "borderBottomStyle": "solid",
      "borderBottomWidth": "0px",
      "borderLeftStyle": "solid",
      "borderLeftWidth": "0px",
      "borderRightStyle": "solid",
      "borderRightWidth": "0px",
      "borderTopStyle": "solid",
      "borderTopWidth": "0px",
      "boxSizing": "border-box",
      "color": "inherit",
      "display": null,
      "flexBasis": "auto",
      "flexDirection": "column",
      "flexShrink": 0,
      "font": "inherit",
      "listStyle": "none",
      "marginBottom": "0px",
      "marginLeft": "0px",
      "marginRight": "0px",
      "marginTop": "0px",
      "minHeight": "0px",
      "minWidth": "0px",
      "msFlexAlign": "stretch",
      "msFlexDirection": "column",
      "msFlexNegative": 0,
      "msPreferredSize": "auto",
      "paddingBottom": "0px",
      "paddingLeft": "0px",
      "paddingRight": "0px",
      "paddingTop": "0px",
      "position": "relative",
      "textAlign": "inherit",
      "textDecoration": "none",
    }
  }
  type="button">
  <span
    className=" __style_df"
    style={
      Object {
        "MozBoxSizing": "border-box",
        "WebkitAlignItems": "stretch",
        "WebkitBoxAlign": "stretch",
        "WebkitBoxDirection": "normal",
        "WebkitBoxOrient": "vertical",
        "WebkitFlexBasis": "auto",
        "WebkitFlexDirection": "column",
        "WebkitFlexShrink": 0,
        "alignItems": "stretch",
        "backgroundColor": "transparent",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "0px",
        "borderLeftStyle": "solid",
        "borderLeftWidth": "0px",
        "borderRightStyle": "solid",
        "borderRightWidth": "0px",
        "borderTopStyle": "solid",
        "borderTopWidth": "0px",
        "boxSizing": "border-box",
        "color": "inherit",
        "display": null,
        "flexBasis": "auto",
        "flexDirection": "column",
        "flexShrink": 0,
        "font": "inherit",
        "listStyle": "none",
        "marginBottom": "0px",
        "marginLeft": "0px",
        "marginRight": "0px",
        "marginTop": "0px",
        "minHeight": "0px",
        "minWidth": "0px",
        "msFlexAlign": "stretch",
        "msFlexDirection": "column",
        "msFlexNegative": 0,
        "msPreferredSize": "auto",
        "paddingBottom": "0px",
        "paddingLeft": "0px",
        "paddingRight": "0px",
        "paddingTop": "0px",
        "position": "relative",
        "textAlign": "inherit",
        "textDecoration": "none",
      }
    } />
</button>
`;
